src/core_plugins/getting_started/public/lib/add_setup_work.js

```diff
@@ -0,0 +1,81 @@
+import { get } from 'lodash';
+import uiRoutes from 'ui/routes';
+import uiChrome from 'ui/chrome';
+import KbnUrlProvider from 'ui/url';
+import { Notifier } from 'ui/notify/notifier';
+import { IndexPatternsGetIdsProvider } from 'ui/index_patterns/_get_ids';
+import { hasOptedOutOfGettingStarted, optOutOfGettingStarted } from 'ui/getting_started/opt_out_helpers';
+
+import {
+  GETTING_STARTED_ROUTE,
+  CREATE_INDEX_PATTERN_ROUTE
+} from './constants';
+
+uiRoutes
+  .addSetupWork(function gettingStartedGateCheck(Private, $injector) {
+    const getIds = Private(IndexPatternsGetIdsProvider);
+    const kbnUrl = Private(KbnUrlProvider);
+    const config = $injector.get('config');
+    const $route = $injector.get('$route');
+
+    const currentRoute = get($route, 'current.$$route');
+
+    return getIds()
+    .then(indexPatterns => {
+      const indexPatternsExist = Array.isArray(indexPatterns) && indexPatterns.length > 0;
+      const isOnGettingStartedPage = get(currentRoute, 'originalPath') === GETTING_STARTED_ROUTE;
+
+      if (indexPatternsExist) {
+
+        // The user need not see the Getting Started page, so opt them out of it
+        optOutOfGettingStarted();
+
+        // Some routes require a default index pattern to be present. If we're
+        // NOT on such a route, there's nothing more to do; send the user on their way
+        if (!currentRoute.requireDefaultIndex) {
+          return;
+        }
+
+        // Otherwise, check if we have a default index pattern
+        let defaultIndexPattern = config.get('defaultIndex');
+
+        // If we don't have an default index pattern, make the first index pattern the
+        // default one
+        if (!Boolean(defaultIndexPattern)) {
+          defaultIndexPattern = indexPatterns[0];
```

Comment:
Yes, that's correct, I decided to change the behavior a bit here.

The scenario where a user has index patterns defined but none of them is a default _should_ never happen. The only way I see it happening is if a user or something else deleted the default index pattern from `.kibana` directly.

Given that, I think the intent of this bit of code should be to "rescue" the user from such a bad scenario. As such I don't know if it matters much which index pattern we choose as a default. That was my thinking behind the new logic.

WDYT?